sql/parser/ast.go

```diff
@@ -704,43 +754,82 @@ func (node *Subquery) String() string {
 	return fmt.Sprintf("(%v)", node.Select)
 }
 
+// BinaryOp represents a binary operator.
+type BinaryOp int
+
+// BinaryExpr.Operator
+const (
+	Bitand BinaryOp = iota
+	Bitor
+	Bitxor
+	Plus
+	Minus
+	Mult
+	Div
+	Mod
+	Concat
+)
+
+var binaryOpName = [...]string{
+	Bitand: "&",
+	Bitor:  "|",
+	Bitxor: "^",
+	Plus:   "+",
+	Minus:  "-",
+	Mult:   "*",
+	Div:    "/",
+	Mod:    "%",
+	Concat: "||",
```

Comment:
In mysql, `||` can be either "concat" or "logical or" depending on the [sql mode](https://dev.mysql.com/doc/refman/5.1/en/expressions.html), defaulting to "or". This alters the precedence of the operator.